### PR TITLE
strInSlice cleanup #12825

### PR DESCRIPTION
**Pull request by pswica**
The cleanup mentioned here:
https://github.com/kubernetes/test-infra/issues/12825

Please let me know if there's anything here I can improve on

### CODE REVIEW

File: prow/plugins/lgtm/lgtm.go [hunk 1]
```diff
@@ -116,24 +116,14 @@ func helpProvider(config *plugins.Configuration, enabledRepos []string) (*plugin
 func optionsForRepo(config *plugins.Configuration, org, repo string) *plugins.Lgtm {
 	fullName := fmt.Sprintf("%s/%s", org, repo)
 	for i := range config.Lgtm {
```
Comment (cblecker) [inline prow/plugins/lgtm/lgtm.go]: Can we replace this loop with the following two loops?

```suggestion
	// First search for repo config
	for _, c := range config.Lgtm {
		if !sets.NewString(c.Repos).Has(fullName) {
			continue
		}
		return &c
	}

	// If you don't find anything, loop again looking for an org config
	for _, c := range config.Lgtm {
		if !sets.NewString(c.Repos).Has(org) {
			continue
		}
		return &c
	}
```

Comment (pswica) [reply prow/plugins/lgtm/lgtm.go]: I think this looks a lot cleaner. I made the change, thanks so much for reviewing

Comment (pswica) [reply prow/plugins/lgtm/lgtm.go]: Also, I ran the update-bazel script before committing. Thanks again I appreciate this a lot.

File: prow/plugins/lgtm/lgtm.go [hunk 2]
```diff
@@ -23,8 +23,8 @@ import (
 	"strings"
 
 	"github.com/sirupsen/logrus"
-	"k8s.io/apimachinery/pkg/util/sets"
 
+	"k8s.io/apimachinery/pkg/util/sets"
```
Comment (fejta) [inline prow/plugins/lgtm/lgtm.go]: k8s.io/apimachinery != k8s.io/test-infra. Please put it in the non-built-in, non-local repo block where it was previously

File: prow/plugins/lgtm/lgtm.go [hunk 3]
```diff
@@ -115,23 +115,20 @@ func helpProvider(config *plugins.Configuration, enabledRepos []string) (*plugin
 // optionsForRepo gets the plugins.Lgtm struct that is applicable to the indicated repo.
 func optionsForRepo(config *plugins.Configuration, org, repo string) *plugins.Lgtm {
 	fullName := fmt.Sprintf("%s/%s", org, repo)
-	for i := range config.Lgtm {
-		if !strInSlice(org, config.Lgtm[i].Repos) && !strInSlice(fullName, config.Lgtm[i].Repos) {
+	for _, c := range config.Lgtm {
+		if !sets.NewString(c.Repos...).Has(fullName) {
```
Comment (stevekuznetsov) [inline prow/plugins/lgtm/lgtm.go]: Make the set once, not on every loop. Might also short-circuit with `!HasAny()`

File: prow/plugins/lgtm/lgtm.go (hunk 2 above)
Comment (pswica) [reply prow/plugins/lgtm/lgtm.go]: Really good point, and thanks a lot for spotting this. I correct in this PR: https://github.com/kubernetes/test-infra/pull/12851 as my last PR was merged before I got home.

File: prow/plugins/lgtm/lgtm.go (hunk 3 above)
Comment (pswica) [reply prow/plugins/lgtm/lgtm.go]: Thanks a lot for spotting this--it definitely seems more efficient and clean. I will correct this in another tiny PR. I really really appreciate your feedback

**Pull request merged**